types/src/transaction/mod.rs

```diff
@@ -969,95 +969,105 @@ impl TransactionOutput {
     }
 }
 
-pub trait TransactionInfoTrait:
-    Clone + CryptoHash + Debug + Display + Eq + Serialize + DeserializeOwned
-{
-    /// Constructs a new `TransactionInfo` object using transaction hash, state root hash and event
-    /// root hash.
-    fn new(
+/// `TransactionInfo` is the object we store in the transaction accumulator. It consists of the
+/// transaction as well as the execution result of this transaction.
+#[derive(Clone, CryptoHasher, BCSCryptoHash, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub enum TransactionInfo {
+    V0(TransactionInfoV0),
+}
+
+impl TransactionInfo {
+    pub fn new(
         transaction_hash: HashValue,
-        state_root_hash: HashValue,
+        state_change_hash: HashValue,
```

Comment:
why change the name?

Reply:
This will be the locally touched account tree root in an alternative strategy, just to remind it's probably not the global state root hash